core/chains/evm/client/send_only_node.go

```diff
@@ -92,6 +104,15 @@ func (s *sendOnlyNode) Start(startCtx context.Context) error {
 	return nil
 }
 
+func (s *sendOnlyNode) SetEthClient(newBatchSender BatchSender, newSender TxSender) {
+	if s.sender != nil {
+		s.log.Warn("SetEthClient should only be called once!  Ignoring")
```

Comment:
I wonder if this should be `Critical` level, or even `Panic`, since it would indicate severe programmer error, no? :thinking:

Reply:
+1